emData/generate_TP.py

```diff
@@ -123,6 +166,13 @@ class ProjoutIndexDisk(Enum):
       "#ifndef TopFunctions_TrackletProcessor_parameters_h\n"
       "#define TopFunctions_TrackletProcessor_parameters_h\n"
       "\n"
+      "#ifndef __SYNTHESIS__\n"
+      '#include "SWLUTReader.h"\n'
+      '#ifdef CMSSW_GIT_HASH\n'
+      '#include <mutex>\n'
+      'std::mutex getLUTMutex_;\n'
```

Comment:
Rather than declaring "std::mutex getLUTMutex_;" as a global variable in the header file, which makes it hard to restrict its scope, I'd recommend declaring it as a static variable inside the function where it is used. i.e. getLUT(). See https://stackoverflow.com/questions/14106653/are-function-local-static-mutexes-thread-safe .